main.go

```diff
@@ -17,35 +19,46 @@ var (
 	// less confusion.
 	storageType = getopt("STORAGE_ADAPTER", "memory")
 	numLines, _ = strconv.Atoi(getopt("NUMBER_OF_LINES", "1000"))
+	nsqHost     = getopt("DEIS_NSQD_SERVICE_HOST", "")
+	nsqPort     = getopt("DEIS_NSQD_SERVICE_PORT_TRANSPORT", "4150")
+	nsqURL      = fmt.Sprintf("%s:%s", nsqHost, nsqPort)
 )
 
 func main() {
-	syslogishServer, err := syslogish.NewServer(storageType, numLines)
+	storageAdapter, err := storage.NewAdapter(storageType, numLines)
 	if err != nil {
-		log.Fatal("Error creating syslogish server", err)
+		log.Fatal("Error creating storage adapter:", err)
 	}
-	weblogServer, err := weblog.NewServer(syslogishServer)
+	logger, err := logs.NewLogger(storageAdapter)
+	if err != nil {
+		log.Fatal("Error creating logger", err)
+	}
+
+	weblogServer, err := weblog.NewServer(logger)
 	if err != nil {
 		log.Fatal("Error creating weblog server", err)
 	}
 
-	syslogishServer.Listen()
 	weblogServer.Listen()
-
 	log.Println("deis-logger running")
 
-	// No cleanup is needed upon termination.  The signal to reopen log files (after hypothetical
-	// logroation, for instance), if applicable, is the only signal we'll care about.  Our main loop
-	// will just wait for that signal.
-	reopen := make(chan os.Signal, 1)
-	signal.Notify(reopen, syscall.SIGUSR1)
-
-	for {
-		<-reopen
-		if err := syslogishServer.ReopenLogs(); err != nil {
-			log.Fatal("Error reopening logs", err)
+	consumer.Register("logs", "consume", 30, func(message *consumer.Message) {
+		err := logger.WriteLog(message.Body)
+		if err != nil {
+			log.Printf("Unable to store message:%v\n%s", err, string(message.Body))
```

Comment:
This is a spot where I'd be very, very careful. From previous experience working on logger, there is a phenomenon I called "the logger death spiral" that you want to avoid. It's an infinite and infinitely _worsening_ loop that results from the logger itself logging some failure, which itself is a log record captured by fluentd and forwarded (via nsq now, of course) back _to_ the logger, where the same condition that triggered the initial error may trigger another. Things can snowball out of control. If there is any option to exempt the logger from having its own logs collected by fluentd, we may even wish to pursue that.